frontend/app_flowy/lib/workspace/presentation/settings/widgets/settings_file_exporter_widget.dart

```diff
@@ -0,0 +1,209 @@
+          },
+        ),
+        const HSpace(8),
+        FlowyTextButton(
+          LocaleKeys.button_OK.tr(),
+          onPressed: () async {
+            // TODO: Export Data
+            await FilePicker.platform.getDirectoryPath().then((exportPath) {
+              Navigator.of(context).pop();
+            });
+          },
+        ),
+      ],
+    );
+  }
+
+  FutureBuilder<dartz.Either<WorkspaceSettingPB, FlowyError>>
+      _buildFileSelector(BuildContext context) {
+    return FutureBuilder<dartz.Either<WorkspaceSettingPB, FlowyError>>(
+      future: FolderEventReadCurrentWorkspace().send(),
+      builder: (context, snapshot) {
+        if (snapshot.hasData &&
+            snapshot.connectionState == ConnectionState.done) {
+          final workspaces = snapshot.data?.getLeftOrNull<WorkspaceSettingPB>();
+          if (workspaces != null) {
+            return _ExpandedList(
+              apps: workspaces.workspace.apps.items,
+              onChanged: (selectedPages) {
+                // _selectedPages = selectedPages;
+              },
+            );
+          }
+        }
+        return const CircularProgressIndicator();
+      },
+    );
+  }
+}
+
+class _ExpandedList extends StatefulWidget {
+  const _ExpandedList({
+    Key? key,
+    required this.apps,
+    required this.onChanged,
+  }) : super(key: key);
+
+  final List<AppPB> apps;
+  final void Function(Map<String, List<String>> selectedPages) onChanged;
+
+  @override
+  State<_ExpandedList> createState() => __ExpandedListState();
+}
+
+class __ExpandedListState extends State<_ExpandedList> {
+  List<AppPB> get apps => widget.apps;
+  List<bool> expanded = [];
+  List<bool> selectedApps = [];
+  List<List<bool>> selectedItems = [];
+
+  @override
+  void initState() {
+    super.initState();
+
+    expanded = apps.map((e) => true).toList();
+    selectedApps = apps.map((e) => true).toList();
+    selectedItems =
+        apps.map((e) => e.belongings.items.map((e) => true).toList()).toList();
+    widget.onChanged(_getSelectedPages());
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _buildExpandedList(context);
+  }
+
+  Widget _buildExpandedList(BuildContext context) {
+    List<Widget> children = [];
+    for (var i = 0; i < apps.length; i++) {
+      children.add(_buildExpandedItem(context, i));
+    }
+    return Material(
+      color: Colors.transparent,
+      child: SingleChildScrollView(
+        child: Column(
+          children: children,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildExpandedItem(BuildContext context, int index) {
+    final isExpaned = expanded[index] == true;
+    List<Widget> expandedChildren = [];
+    if (isExpaned) {
+      for (var i = 0; i < selectedItems[index].length; i++) {
+        final name = apps[index].belongings.items[i].name;
+        final checkbox = CheckboxListTile(
+          value: selectedItems[index][i],
+          onChanged: (value) {
+            setState(() {
```

Comment:
I think we should use `bloc` or `cubic` to keep the Widget clear. Then some of the logic of  `_getSelectedPages` and  `initState ` can be removed.

Reply:
I separated the logic into a new file called `file_exporter_cubit.dart`.